posit-dev/positron
# Add additional env vars for more metadata about the Positron session

Addresses #4082 

### TODO

- [x] Bump ark with https://github.com/posit-dev/ark/pull/531

### QA Notes

This PR adds some new env vars to specify the mode (desktop vs. web) and a long version number, as well as making the terminal env vars more consistent with the console. You can find them via:

- the R console

```r
Sys.getenv("POSITRON_MODE")
Sys.getenv("POSITRON_LONG_VERSION")
```

- the Python console

```python
import os
print(os.environ['POSITRON_MODE'])
print(os.environ['POSITRON_LONG_VERSION'])
```

- terminals in Positron

```bash
echo "$POSITRON_MODE"
echo "$POSITRON_VERSION"
echo "$POSITRON_LONG_VERSION"
```

In a dev build, the build number isn't defined so you'll see something like "2024.03.0+undefined" on a web build or "2024.09.0+0" on a desktop build.
